usr/src/lib/libzfs/common/libzfs_dataset.c

```diff
@@ -3660,7 +3660,7 @@ zfs_promote(zfs_handle_t *zhp)
 			return (zfs_error(hdl, EZFS_EXISTS, errbuf));
 
 		default:
-			return (zfs_standard_error(hdl, save_errno, errbuf));
+			return (zfs_standard_error(hdl, ret, errbuf));
```

Comment:
@andy-js @ahrens I think this change is needed, in addition to the earlier changes; right? I see other functions pass in `errno`, but I think that's incorrect for the same reason that we removed `save_errno` to begin with. Let me know what you think.